### PR TITLE
ballle98/AqualinkD#53: In PDA mode pressing spa heat enables spa mode…

### LINKED ISSUE #137: In PDA mode pressing spa heat enables spa mode instead
see ballle98/AqualinkD#53

request for button "spa_heater" was matching button label "spa" because the sting comparison was only going to the length of the label.  Strings need to match in length and in content.  This is similar to an older PDA issue ballle98/AqualinkD#1

```
11:10:24.507 Debug:   NetService:WS: Message - Key 'command' Value 'Spa_Heater'
11:10:24.509 Debug:   NetService:WS: Message - Key 'uri' Value 'Spa_Heater/set'
11:10:24.512 Debug:   NetService:WS: Message - Key 'value' Value '1'
11:10:24.513 Debug:   NetService:WS: URI Request 'Spa_Heater/set': value 1.00
11:10:24.514 Info:    NetService:WS: received 'ON' for 'Spa_Mode', turning 'ON'
11:10:24.518 Debug:   Panl Prog: Thread 13,0xb5401f90 is active (Switch PDA device on/off)
```

### PR DESCRIPTION
**Pull request by ballle98**
fix #137

### CODE REVIEW

**Comment by ballle98** [comment]
@sfeakes updated this PR to 2.3.0f

**Comment by ballle98** [comment]
@sfeakes ??? not sure what your looking at go here https://github.com/sfeakes/AqualinkD/pull/138/files 1 file changed 11 lines added 5 lines deleted

**Comment by sfeakes** [comment]
Sorry, I was looking at the diff in the message chain.
https://github.com/sfeakes/AqualinkD/compare/e3648a60f754bed384b03aede3ed18f85e692e6c..2d970b227b5e4eb976e0dcbd2ce1e35e757fd7bf

Can you please provide details on how to replicate this problem, I've tried and can't reproduce.  Line 1143 in net_services.c should have fixed this problem a long time ago.

**Comment by ballle98** [comment]
see issue #137 

```
11:10:24.507 Debug:   NetService:WS: Message - Key 'command' Value 'Spa_Heater'
11:10:24.509 Debug:   NetService:WS: Message - Key 'uri' Value 'Spa_Heater/set'
11:10:24.512 Debug:   NetService:WS: Message - Key 'value' Value '1'
11:10:24.513 Debug:   NetService:WS: URI Request 'Spa_Heater/set': value 1.00
11:10:24.514 Info:    NetService:WS: received 'ON' for 'Spa_Mode', turning 'ON'
11:10:24.518 Debug:   Panl Prog: Thread 13,0xb5401f90 is active (Switch PDA device on/off)

```

**Comment by sfeakes** [comment]
What version are you using?

With the below set in the config
`button_02_label=Spa`

This is the output
```
Debug: NetService:WS: Message - Key 'command' Value 'Spa_Heater'
Debug: NetService:WS: Message - Key 'uri' Value 'Spa_Heater/set'
Debug: NetService:WS: Message - Key 'value' Value '1'
Debug: NetService:WS: URI Request 'Spa_Heater/set': value 1.00
Info: AqualinkD: received 'ON' for 'Spa_Heater', turning 'ON'
```

**Comment by ballle98** [comment]
OK the code has definately changed since I saw this issue.  This log

`11:10:24.514 Info:    NetService:WS: received 'ON' for 'Spa_Mode', turning 'ON'`

was coming from line below but this is commented code

https://github.com/sfeakes/AqualinkD/blob/4a6c948a6aff1b6e634e2f0497416655315b61c1/aq_panel.x.c#L650

File: net_services.c [hunk 1]
```diff
@@ -1055,10 +1059,12 @@ uriAtype action_URI(request_source from, const char *URI, int uri_length, float
...
     }
 
+    snprintf(labelBuff, sizeof(labelBuff), "%.*s", ri2-ri1-1, ri1);
     for (i=0; i < _aqualink_data->total_buttons && found==false; i++) {
       // If Label = "Spa", "Spa_Heater" will turn on "Spa", so need to check '/' on label as next character
-      if (strncmp(ri1, _aqualink_data->aqbuttons[i].name, strlen(_aqualink_data->aqbuttons[i].name)) == 0 ||
-          (strncmp(ri1, _aqualink_data->aqbuttons[i].label, strlen(_aqualink_data->aqbuttons[i].label)) == 0 && ri1[strlen(_aqualink_data->aqbuttons[i].label)] == '/'))
+      if ((strcmp(labelBuff, _aqualink_data->aqbuttons[i].name) == 0) ||
+          ((strcmp(labelBuff, _aqualink_data->aqbuttons[i].label) == 0) &&
+           (ri1[strlen(_aqualink_data->aqbuttons[i].label)] == '/')))
```
Comment (ballle98) [inline net_services.c:1067]: When this issue was opened the check for next characeter == '/' was not there.

**Pull request closed without merging**